packages/http-client-csharp/src/lib/model.ts

```diff
@@ -0,0 +1,1019 @@
+import {
+  EncodeData,
+  Enum,
+  EnumMember,
+  IntrinsicType,
+  Model,
+  ModelProperty,
+  Namespace,
+  NeverType,
+  Operation,
+  Program,
+  Scalar,
+  Type,
+  Union,
+  UsageFlags,
+  getDeprecated,
+  getDiscriminator,
+  getDoc,
+  getEffectiveModelType,
+  getEncode,
+  getFormat,
+  getKnownValues,
+  getVisibility,
+  isArrayModelType,
+  isGlobalNamespace,
+  isRecordModelType,
+  isVoidType,
+  navigateTypesInNamespace,
+  resolveUsages,
+} from "@typespec/compiler";
+import {
+  HttpOperation,
+  getHeaderFieldName,
+  getPathParamName,
+  getQueryParamName,
+  isStatusCode,
+} from "@typespec/http";
+import { getResourceOperation } from "@typespec/rest";
+import { NetEmitterOptions } from "../options.js";
+import { fromSdkEnumType } from "../type/converter.js";
+import { FormattedType } from "../type/formattedType.js";
+import { InputEnumTypeValue } from "../type/inputEnumTypeValue.js";
+import { InputIntrinsicTypeKind } from "../type/inputIntrinsicTypeKind.js";
+import { InputModelProperty } from "../type/inputModelProperty.js";
+import { InputPrimitiveTypeKind } from "../type/inputPrimitiveTypeKind.js";
+import {
+  InputDictionaryType,
+  InputEnumType,
+  InputIntrinsicType,
+  InputListType,
+  InputLiteralType,
+  InputModelType,
+  InputPrimitiveType,
+  InputType,
+  InputUnionType,
+  isInputDictionaryType,
+  isInputEnumType,
+  isInputIntrinsicType,
+  isInputListType,
+  isInputLiteralType,
+  isInputModelType,
+} from "../type/inputType.js";
+import { InputTypeKind } from "../type/inputTypeKind.js";
+import { LiteralTypeContext } from "../type/literalTypeContext.js";
+import { Usage } from "../type/usage.js";
+import { logger } from "./logger.js";
+import { capitalize, getFullNamespaceString, getTypeName } from "./utils.js";
+/**
+ * Map calType to csharp InputTypeKind
+ */
+export function mapTypeSpecTypeToCSharpInputTypeKind(
+  context: SdkContext,
+  typespecType: Type,
+  format?: string,
+  encode?: EncodeData
+): InputPrimitiveTypeKind {
+  const kind = typespecType.kind;
+  switch (kind) {
+    case "Model":
+      return getCSharpInputTypeKindByIntrinsicModelName(typespecType.name, format, encode);
+    case "ModelProperty":
+      return InputPrimitiveTypeKind.Object;
+    case "Enum":
+      return InputPrimitiveTypeKind.Enum;
+    case "Number":
+      let numberValue = typespecType.value;
+      if (numberValue % 1 === 0) {
+        return InputPrimitiveTypeKind.Int32;
+      }
+      return InputPrimitiveTypeKind.Float64;
+    case "Boolean":
+      return InputPrimitiveTypeKind.Boolean;
+    case "String":
+      if (format === "date") return InputPrimitiveTypeKind.DateTime;
+      if (format === "uri") return InputPrimitiveTypeKind.Uri;
+      return InputPrimitiveTypeKind.String;
+    default:
+      return InputPrimitiveTypeKind.UnKnownKind;
+  }
+}
+
+function getCSharpInputTypeKindByIntrinsicModelName(
+  name: string,
+  format?: string,
+  encode?: EncodeData
+): InputPrimitiveTypeKind {
```

Comment:
This might benefit from narrowing the allowed inputs using the existing collection of intrinsic scalar names, as here: https://github.com/markcowl/cadl/blob/a8f1249b8bf1c0370e993ce1ec856232b692ee19/packages/service-generator-csharp/src/utils.ts#L355

Reply:
Can you elaborate on what the issue might be with the existing implementation?  This is existing code that I just copied so I don't have a lot context here.